model/minsky.cc

```diff
@@ -713,6 +725,103 @@ namespace minsky
        });
     canvas.requestRedraw();
   }
+  
+  /*
+  For using GSL non-linear least square fitting routines
+  patterned on first Large Nonlinear Least Squares Example.
+  See online documentation at https://www.gnu.org/software/gsl/doc/html/nls.html
+  */
+      
+namespace
+{	
+
+   int residual_func(const gsl_vector * x, void *params,
+          gsl_vector * f)
+   {
+	if (params==NULL) return GSL_EBADFUNC;
+	else { 	  
+		
+	  struct OPTpars *par = ((struct OPTpars *)params); 
+	      
+          int err = GSL_SUCCESS;
+          auto n = ValueVector::stockVars.size();
+          vector<double> xx(n);
+          
+          for (size_t i = 0; i < n; i++)
+             xx[i] = gsl_vector_get(x,i);     
+          
+          err=gsl_odeiv2_driver_apply(par->d->driver, par->t,numeric_limits<double>::max(),&xx[0]);
+	      
+	  // define residual vector function to be optimized 
+          for (size_t i = 0; i < n; i++)
+              gsl_vector_set (f, i, xx[i] - par->x_init[i]);  
+            
+          switch (err)
+              {
+              case GSL_SUCCESS: case GSL_EMAXITER: return GSL_SUCCESS;
+              case GSL_FAILURE:
+                throw error("unspecified error GSL_FAILURE returned");
+              case GSL_EBADFUNC: 
+                gsl_odeiv2_driver_reset(par->d->driver);
+                throw error("Invalid arithmetic operation detected");
+              default:
+                throw error("gsl error: %s",gsl_strerror(err));
+              }          
+          
+      }
+   
```

Comment:
It looks like you are missing a close brace here, for terminating the body of residual_func. I think that is why your compiles are failing.

Also note that you are doing an early return on line 741, so the else statement on line 742 is not necessary and only serves to increase the indent level unnecessarily (not good).

Reply:
Thank you, I have fixed this.